app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,193 @@
+package com.example.drawingactivity
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.drawingactivity.DialogUtils.InputLimitations.CHAR_LIMIT
+import com.example.drawingactivity.DialogUtils.InputLimitations.LAYER_COUNT_START
+import com.example.drawingactivity.DialogUtils.InputLimitations.NAME_EMPTY
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogUtils {
+
+    private var hintIndex: Int = LAYER_COUNT_START
+
+    fun nameLayerDialog(
+        context: Context,
+        layerList: MutableList<String>,
+        /**
+         * Called whenever text is changed in the editable field.
+         * This uses a string and returns nothing.
+         */
+        onEditText: ((String) -> Unit)?,
+        /**
+         * Called when the Dialog is building.
+         * Checks to see if there is a String available and returns it.
+         * If there is none, it will return null.
+         */
+        onCheckForInput: (() -> String?)?,
+        /**
+         * This is Called when there is an Error and the Positive Button is pressed on the Dialog.
+         * This toasts an appropriate error for why the layer couldn't be added.
+         */
+        onFailure: ((String) -> Unit)?,
+        /**
+         * This is Called when there is no Error and the Positive Button is pressed on the Dialog.
+         * This adds a Layer to a list.
+         */
+        onNameLayer: ((String) -> Unit)?,
+        /**
+         * Called when the Dialog Completes through any listener.
+         * This will reset all values to null in preparation for the next opening of the Dialog.
+         */
+        onDialogComplete: (() -> Unit)?,
```

Comment:
I think having it exit using `onSubmit` with a string that's null (invalid) or not null (valid) makes this function obsolete. Let's trim this signature down